### PR TITLE
Added support for custom EncoderDecoder models

### LINKED ISSUE #670: Error while loading a pretrained EncoderDecoder model
Hi all,
I've trained an `EncoderDecoder` model but I cannot load it via `AutoModelForSeq2SeqLMWithValueHead.from_pretrained`.

My `EncoderDecoder` model has been defined by tying together two RoBERTa model, as follows:

```python
from transformers import EncoderDecoderModel

pretrained_encoder = "seyonec/ChemBERTa-zinc-base-v1"
bert2bert = EncoderDecoderModel.from_encoder_decoder_pretrained(
    pretrained_encoder,
    pretrained_encoder,
    tie_encoder_decoder=True,
)
```

Transformers and TRL versions:

```
trl.__version__: 0.5.0
transformers.__version__: 4.31.0
```

Code to reproduce the issue:

```python
import torch
from transformers import (
    EncoderDecoderConfig,
    AutoTokenizer,
)
from trl import (
    AutoModelForSeq2SeqLMWithValueHead,
    PPOConfig,
    PPOTrainer,
    create_reference_model,
)

pretrained_model = "ribesstefano/ChemBERTa2ChemBERTa-58M"
model = AutoModelForSeq2SeqLMWithValueHead.from_pretrained(pretrained_model)
model_ref = create_reference_model(model, num_shared_layers=6)

tokenizer = AutoTokenizer.from_pretrained(pretrained_model)
tokenizer.pad_token = tokenizer.eos_token
```

Error message:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-45-7d8719dacb7f>](https://localhost:8080/#) in <cell line: 22>()
     20 # print(bert2bert.config.hidden_size)
     21 
---> 22 model = AutoModelForSeq2SeqLMWithValueHead.from_pretrained(pretrained_model)
     23 # # model_ref = AutoModelForSeq2SeqLMWithValueHead.from_pretrained(pretrained_model, device_map="auto", load_in_8bit=True)
     24 # model_ref = create_reference_model(model, num_shared_layers=6)

[/usr/local/lib/python3.10/dist-packages/trl/models/modeling_base.py](https://localhost:8080/#) in from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs)
    230                     raise ValueError("PromptLearningConfig is not supported for PPO training.")
    231         # Then, create the full model by instantiating the wrapper class
--> 232         model = cls(pretrained_model, **trl_model_args)
    233 
    234         # if resume_training, load the state_dict again - this is ok since the

[/usr/local/lib/python3.10/dist-packages/trl/models/modeling_value_head.py](https://localhost:8080/#) in __init__(self, pretrained_model, **kwargs)
    289             raise ValueError("The model does not have a language model head, please use a model that has one.")
    290 
--> 291         self.v_head = ValueHead(self.pretrained_model.config, **v_head_kwargs)
    292 
    293         self._init_weights(**v_head_kwargs)

[/usr/local/lib/python3.10/dist-packages/trl/models/modeling_value_head.py](https://localhost:8080/#) in __init__(self, config, **kwargs)
     37             hidden_size = config.word_embed_proj_dim
     38         else:
---> 39             hidden_size = config.hidden_size
     40 
     41         self.summary = nn.Linear(hidden_size, 1)

[/usr/local/lib/python3.10/dist-packages/transformers/configuration_utils.py](https://localhost:8080/#) in __getattribute__(self, key)
    259         if key != "attribute_map" and key in super().__getattribute__("attribute_map"):
    260             key = super().__getattribute__("attribute_map")[key]
--> 261         return super().__getattribute__(key)
    262 
    263     def __init__(self, **kwargs):

AttributeError: 'EncoderDecoderConfig' object has no attribute 'hidden_size'
```

From what I can guess from the source code, `ValueHead` only inspect the model's top config, expecting an `hidden_size`. However, that information is in the `model.decoder.config.hidden_size` for an `EncoderDecoder` class. Does it sound like a bug or something to update in the code? Or is there any workaround I can try?

Thanks in advance

### PR DESCRIPTION
**Pull request by ribesstefano**
Resolving issue #670.

Fixes #670

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Works for me, thanks!

**Comment by younesbelkada** [comment]
Actually let's wait for the CI to be green before merging

**Comment by lvwerra** [comment]
Can you merge main into your branch? We fixed the flaky tests in the CI.

**Pull request merged**
